# Changes the logging code to use the correct version of the syslog call.
# This is a necessary change as using the wrong version of the syslog call
# results in segmentation faults when attempting to produce logs.
#
# This patch has already been sent upstream and merged, but there is currently
# no tagged release which uses this change so this patch is only temporary and
# will be removed once a new release is created.
#
# The merge to the upstream repository can be viewed here:
# https://github.com/OpenPrinting/ippusbxd/pull/1

diff --git a/src/logging.c b/src/logging.c
index e60beb9..d66d304 100644
--- a/src/logging.c
+++ b/src/logging.c
@@ -32,7 +32,7 @@ void BASE_LOG(enum log_level level, const char *fmt, ...)
   if (g_options.log_destination == LOGGING_STDERR)
     vfprintf(stderr, fmt, arg);
   else if (g_options.log_destination == LOGGING_SYSLOG)
-    syslog(LOG_ERR, fmt, arg);
+    vsyslog(LOG_ERR, fmt, arg);
   va_end(arg);
 }
 
